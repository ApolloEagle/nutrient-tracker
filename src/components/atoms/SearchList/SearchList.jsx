import { List, ListItem, ListItemText, ListItemButton } from "@mui/material";

function SearchList({ list, dense, handleSelect }) {
  return (
    <List
      sx={{
        overflow: "auto",
        maxHeight: 100,
        zIndex: 1,
        position: "absolute",
        backgroundColor: "white",
        opacity: 0.95,
        width: "40.5%",
      }}
    >
      {list?.map((item, index) => {
        return (
          <ListItemButton dense={dense} key={index}>
            <ListItem onClick={() => handleSelect(item)}>
              <ListItemText primary={item.description} />
            </ListItem>
          </ListItemButton>
        );
      })}
    </List>
  );
}

export default SearchList;
