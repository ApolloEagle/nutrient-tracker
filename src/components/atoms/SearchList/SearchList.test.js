import React from 'react';
import { shallow } from 'enzyme';
import SearchList from './SearchList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const wrapper = shallow(<SearchList />);

describe('Search List', () => {
  it('Renders search list', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
