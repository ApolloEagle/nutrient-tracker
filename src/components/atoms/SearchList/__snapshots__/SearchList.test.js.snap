// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search List Renders search list 1`] = `
<ForwardRef(List)
  sx={
    Object {
      "backgroundColor": "white",
      "maxHeight": 100,
      "opacity": 0.95,
      "overflow": "auto",
      "position": "absolute",
      "width": "40.5%",
      "zIndex": 1,
    }
  }
/>
`;
